docs/source/format/Integration.rst

```diff
@@ -20,8 +20,92 @@
 Integration Testing
 ===================
 
+Our strategy for integration testing between Arrow implementations is:
+
+* Test datasets are specified in a custom human-readable, JSON-based format
+  designed for Arrow
+* Each implementation provides a testing executable capable of converting
+  between the JSON and the binary Arrow file representation
+* The test executable is also capable of validating the contents of a binary
+  file against a corresponding JSON file
+
+Running integration tests
+-------------------------
+
+The integration test data generator and runner uses ``archery``, a Python script
+that requires Python 3.6 or higher. You can create a standalone Python
+distribution and environment for running the tests by using
+`miniconda <https://conda.io/miniconda.html>`_. On Linux this is:
+
+.. code-block:: shell
+   MINICONDA_URL=https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
+   wget -O miniconda.sh $MINICONDA_URL
+   bash miniconda.sh -b -p miniconda
+   export PATH=`pwd`/miniconda/bin:$PATH
+
+   conda create -n arrow-integration python=3.6 nomkl numpy six
+   conda activate arrow-integration
+
+
+If you are on macOS, instead use the URL:
+
+.. code-block:: shell
+   MINICONDA_URL=https://repo.continuum.io/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
+
+Once you have Python, you can install archery
+
+.. code-block:: shell
+   pip install -e dev/archery
+
+The integration tests are run using the ``archery integration`` command.
+
+.. code-block:: shell
+   archery integration --help
+
+In order to run integration tests, you'll first need to build each component
+you want to include. See the respective developer docs for C++, Java, etc.
+for instructions on building those.
+
+Some languages may require additional build options to enable integration
+testing. For C++, for example, you need to add ``-DARROW_BUILD_INTEGRATION=ON``
+to your cmake command.
+
+Depending on which components you have built, you can enable and add them to
+the archery test run. For example, if you only have the C++ project built, run:
+
+.. code-block:: shell
+   archery integration --with-cpp=1
+
+
+For Java, it may look like:
+
+.. code-block:: shell
+   VERSION=0.11.0-SNAPSHOT
+   export ARROW_JAVA_INTEGRATION_JAR=$JAVA_DIR/tools/target/arrow-tools-$VERSION-jar-with-dependencies.jar
+   archery integration --with-cpp=1 --with-java=1
+
+To run all tests, including Flight integration tests, do:
+
+.. code-block:: shell
+   archery integration --with-all --run-flight
+
+Note that we run these tests in continuous integration, and the CI job uses
+docker-compose. You may also run the docker-compose job locally, or at least
+refer to it if you have questions about how to build other languages or enable
+certain tests.
```

Comment:
A link to the latest information about how to find out how to run the docker-compose jobs would be good here.

Reply:
Will do, though _those_ docs aren't in great shape either.